src/sql/src/plan/statement/ddl.rs

```diff
@@ -1578,15 +1377,41 @@ pub fn plan_create_subsource(
             ProtoSourceExportStatementDetails::decode(&*details).map_err(|e| sql_err!("{}", e))?;
         let details =
             SourceExportStatementDetails::from_proto(details).map_err(|e| sql_err!("{}", e))?;
-        // TODO(roshan): Use these options in the IngestionExport struct below. For now we just
-        // decode here to ensure that the details are present and valid.
-        let _ = details;
-        let _ = text_columns;
-        let _ = ignore_columns;
-
+        let details = match details {
+            SourceExportStatementDetails::Postgres { table } => {
+                SourceExportDetails::Postgres(PostgresSourceExportDetails {
+                    table_cast: crate::pure::postgres::generate_table_cast(
+                        scx,
+                        &table,
+                        &text_columns,
+                    )?,
+                    table,
+                })
+            }
+            SourceExportStatementDetails::MySql {
+                table,
+                initial_gtid_set,
+            } => SourceExportDetails::MySql(MySqlSourceExportDetails {
+                table,
+                initial_gtid_set,
+                text_columns: text_columns.into_iter().map(|c| c.into_string()).collect(),
+                ignore_columns: ignore_columns
+                    .into_iter()
+                    .map(|c| c.into_string())
+                    .collect(),
+            }),
+            SourceExportStatementDetails::LoadGenerator => {
+                SourceExportDetails::LoadGenerator(LoadGeneratorSourceExportDetails {
```

Comment:
I've refactored this in the latest commit to try and address your concern -- it now validates the database and schema as well as the table name in a new `external_reference_to_output` method defined close to where the load-gen database and schema-names come from.

> As part of planning we need a generic method that takes a list of the available external references from the source in fully-qualified form and a list of potentially not fully qualified desired references (the thing the user typed in FOR TABLES ...) and returns the list of fully qualified unambiguous references that are then used throughout the codebase.

The logic that you are referencing already half exists for load-generator sources as part of purification -- the external-references are generated here as 'available subsources' based on the load-gen views:
https://github.com/MaterializeInc/materialize/blob/d973dbce1e99bc777f001bc478264cdf1d8cb93a/src/sql/src/plan/statement/ddl.rs#L1822-L1835

The second half will be something I have to introduce for planning of the `CREATE TABLE .. FROM SOURCE` statements since we need to find the appropriate fully-qualified external reference based on the user-provided reference.
However I'd rather not refactor the subsource statement generation stuff again for load-gen sources since those will be deprecated soon and would require another catalog migration.

My intent with the new `LoadGeneratorOutput` enum is to be the unambiguous way to reference an upstream load-generator object, just like `MySqlTableDesc` and `PostgresTableDesc` are used to find the appropriate upstream table to ingest from mysql/postgres (the external references `UnresolvedItemName`s aren't used anywhere in source-rendering)